src/main/java/uk/gov/pay/ledger/agreement/entity/AgreementsFactory.java

```diff
@@ -32,4 +37,18 @@ public PaymentInstrumentEntity createPaymentInstrument(EventDigest eventDigest)
         entity.setEventCount(eventDigest.getEventCount());
         return entity;
     }
+
+    private AgreementStatus getStatus(EventDigest eventDigest) {
+        var eventType = eventDigest.getMostRecentSalientEventType().get();
+
+        return EVENT_TYPE_AGREEMENT_STATE_MAP.get(eventType);
+    }
+
+    private static final Map<SalientEventType, AgreementStatus> EVENT_TYPE_AGREEMENT_STATE_MAP =
```

Comment:
I think it would be better for this static map to live in the `AgreementStatus` enum so that it mirrors `TransactionState` and `PayoutState` to keep everything a bit more uniform.

Reply:
`AgreementStatus` comes from `pay-java-commons`, which doesn't know about `SalientEventType`.  I'm open to suggestions for where else the map could live though.

Reply:
Aha, I didn't spot that! In that case I think it is fine for this to live here. I guess the only minor change I might make would be to move this towards the top of the class where we usually define constants.